apps/src/templates/studentSnapshot/StudentSnapshot.tsx

```diff
@@ -1,12 +1,68 @@
 import {Typography} from '@mui/material';
-import React from 'react';
+import React, {useEffect, useState} from 'react';
+
+import type {
+  Rubric,
+  RubricData,
+  StudentLevelInfo,
+} from '@cdo/apps/types/rubricTypes';
+import HttpClient from '@cdo/apps/util/HttpClient';
 
 import Header from './header';
+import StudentRubricWidget from './StudentRubricWidget';
 import WidgetTemplate from './widgetTemplate';
 
 import styles from './studentSnapshot.module.scss';
 
 const StudentSnapshot: React.FC = () => {
+  // Hardcoded values for testing - replace with actual values from URL/Redux later
+  const HARDCODED_RUBRIC_ID = 1; // Replace with actual rubric ID
+  const HARDCODED_STUDENT_ID = 1; // Replace with actual student ID
+
+  const [rubric, setRubric] = useState<Rubric | null>(null);
+  const [studentLevelInfo, setStudentLevelInfo] =
+    useState<StudentLevelInfo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      setError(null);
+
+      try {
+        // Fetch rubric
```

Comment:
```suggestion
```
nit: completely up to you, but I don't find these comments explaining what the next line does very helpful. I prefer minimal comments except for those that add context that isn't in the file or some complexity that isn't obvious.